docs/source/cpp/acero.rst

```diff
@@ -27,6 +27,12 @@ Acero: A C++ streaming execution engine
 
     Acero is experimental and a stable API is not yet guaranteed.
 
+.. note::
+
+    If you are interested in contributing to Acero or learning about its
+    internals, please see the `Acero Developer's Guide
+    <../developers/cpp/acero>`_.
```

Comment:
This links to page 404. I think we need an anchor.

Reply:
Like this [`d6d32a8` (#46476)](https://github.com/apache/arrow/pull/46476/commits/d6d32a878ef2b3deedc8d309465ca5293c161c51) ? @amoeba

Reply:
Can we use a `:doc:` directive to link directly to the doc? I'll see if I can get a working local docs build to test.

Reply:
Good point, and it works in my local. Updated in [`a6e27d0` (#46476)](https://github.com/apache/arrow/pull/46476/commits/a6e27d0cd11465ecfaece072987f966d19dacae3). Thank you for the tip!

Reply:
It works as expected in the preview: http://crossbow.voltrondata.com/pr_docs/46476/cpp/acero.html

@amoeba PTAL. Thanks.

Reply:
Yep, looks right. Thanks.